EDU-751: Add caution blurb to StartToCloseTimeout (#2058)
temporalio/documentation: docs/concepts/activities.md

For long running Activity Executions, we recommend also using <a class="tdlp" href="#activity-heartbeat">Activity Heartbeats<span class="tdlpiw"><img src="/img/link-preview-icon.svg" alt="Link preview icon" /></span><span class="tdlpc"><span class="tdlppt">What is an Activity Heartbeat?</span><br /><br /><span class="tdlppd">An Activity Heartbeat is a ping from the Worker that is executing the Activity to the Temporal Cluster. Each ping informs the Temporal Cluster that the Activity Execution is making progress and the Worker has not crashed.</span><span class="tdlplm"><br /><br /><a class="tdlplma" href="#activity-heartbeat">Learn more</a></span></span></a> and <a class="tdlp" href="#heartbeat-timeout">Heartbeat Timeouts<span class="tdlpiw"><img src="/img/link-preview-icon.svg" alt="Link preview icon" /></span><span class="tdlpc"><span class="tdlppt">What is a Heartbeat Timeout?</span><br /><br /><span class="tdlppd">A Heartbeat Timeout is the maximum time between Activity Heartbeats.</span><span class="tdlplm"><br /><br /><a class="tdlplma" href="#heartbeat-timeout">Learn more</a></span></span></a>.

:::caution

> Setting start-to-close or heartbeat timeout is strongly recommended. This is because Temporal does not detect worker crashes directly and rely on these timeouts to force activity retries.

Reviewer: **Please move these changes to `docs-src/concepts/what-is-a-start-to-close-timeout.md` so that they aren't lost when `docs/concepts/activities.md` is generated.**

The following edits conform to our [style guide](https://github.com/temporalio/documentation/blob/main/STYLE.md) and aren't meant to preclude the technical issues raised by @tsurdilo or @cretz.

```suggestion
We strongly recommend setting Start-To-Close Timeout or Heartbeat Timeout.
The Temporal Server does not detect Worker crashes directly and relies on these timeouts to force Activity retries.
```

Author: Thanks @djmagee. I don't know why I thought there is no docs-src for this page.. Updated.
